ui/raidboss/data/05-shb/eureka/zadnor.js

```diff
@@ -81,30 +133,1155 @@ export default {
         for (const key in ceIds) {
           if (ceIds[key] === ceId) {
             if (data.options.Debug)
-              console.log(`Start CE: ${ceId} (${key})`);
-            data.ce = ceId;
+              console.log(`Start CE: ${key} (${ceId})`);
+            data.ce = key;
             return;
           }
         }
 
         if (data.options.Debug)
-          console.log(`Unknown CE: ${ceId}`);
+          console.log(`Start CE: ??? (${ceId})`);
       },
     },
     // ***** On Serpents' Wings *****
+    {
+      id: 'Zadnor Serpents Turbine',
+      netRegex: NetRegexes.startsUsing({ source: 'Stormborne Zirnitra', id: '5E54' }),
+      condition: (data) => data.ce === 'serpents',
+      delaySeconds: (_data, matches) => parseFloat(matches.castTime) - 5,
+      alertText: (data, _matches, output) => {
+        data.serpentsTurbineCount = (data.serpentsTurbineCount || 0) + 1;
+        // TODO: how does this loop?
+        if (data.serpentsTurbineCount === 1)
+          return output.knockbackDonut();
+        else if (data.serpentsTurbineCount === 2)
+          return output.knockbackIntoCircle();
+        else if (data.serpentsTurbineCount === 3)
+          return output.knockbackExplosion();
+        else if (data.serpentsTurbineCount === 4)
+          return output.knockbackDonut();
+        else if (data.serpentsTurbineCount === 5)
+          return output.knockbackIntoSafe();
+      },
+      outputStrings: {
+        knockbackDonut: {
+          en: 'Knockback + Stack Donuts Middle',
+        },
+        knockbackIntoCircle: {
+          en: 'Knockback (towards first circles)',
+        },
+        knockbackIntoSafe: {
+          en: 'Knockback (towards open spots)',
+        },
+        knockbackExplosion: {
+          // Can't trust people to make a safe spot,
+          // so using knockback prevention is probably the best advice.
+          en: 'Knockback (prevent)',
+        },
+      },
+    },
     // ***** Feeling the Burn *****
+    {
+      id: 'Zadnor Feeling Suppressive Magitek Rays',
+      netRegex: NetRegexes.startsUsing({ source: 'Blackburn', id: '5C40', capture: false }),
+      condition: (data) => data.ce === 'feeling',
+      response: Responses.aoe(),
+    },
+    {
+      id: 'Zadnor Feeling Chain Cannon You',
+      netRegex: NetRegexes.headMarker({ id: '00A4' }),
+      condition: (data, matches) => data.ce === 'feeling' && data.me === matches.target,
+      alertText: (_data, _matches, output) => output.text(),
+      outputStrings: {
+        text: {
+          en: 'Chain Cannon on YOU',
```

Comment:
Is this a laser? Or is this similar to the DRN "lasers with duration"?
```suggestion
          en: 'Laser on YOU',
          de: 'Laser auf DIR',
          fr: 'Laser sur VOUS',
          ja: '自分にレーザー',
          cn: '直线激光点名',
          ko: '레이저 대상자',
```